### ISSUE TITLE
Agents: when edit agent tags agent is not subscribed to the group with matching tags

### ISSUE THREAD

**Issue by manrodrigues** [ns1labs/orb#808]
Steps to reproduce:
1 - provision an agent without any tags
2 - create a group
3 - edit agent inserting same tags as group
4 - check container logs

Expected result:
- agent must be successfully subscribed to the group

Current result:
- MQTT error and failed to subscribed
```

{"level":"info","ts":1643381512.8828945,"caller":"pktvisor/pktvisor.go:247","msg":"pktvisor stdout","log":"[2022-01-28 14:51:52.882] [visor] [info] tap [default_pcap]: parsing"}
{"level":"info","ts":1643381512.8829815,"caller":"pktvisor/pktvisor.go:247","msg":"pktvisor stdout","log":"[2022-01-28 14:51:52.882] [visor] [info] tap [default_pcap]: loaded, type pcap"}
{"level":"info","ts":1643381512.8830326,"caller":"pktvisor/pktvisor.go:247","msg":"pktvisor stdout","log":"[2022-01-28 14:51:52.883] [visor] [info] sending anonymous usage metrics (once/day, use --no-track to disable): 0181ad2-poleved-0.4.3.pktvisord.metrics.pktvisor.dev."}
{"level":"info","ts":1643381512.9027624,"caller":"pktvisor/pktvisor.go:247","msg":"pktvisor stdout","log":"[2022-01-28 14:51:52.902] [visor] [info] web server listening on localhost:10853"}
{"level":"info","ts":1643381513.876443,"caller":"pktvisor/pktvisor.go:273","msg":"pktvisor process started","pid":20}
{"level":"info","ts":1643381514.9659307,"caller":"pktvisor/pktvisor.go:247","msg":"pktvisor stdout","log":"[2022-01-28 14:51:54.965] [visor] [info] REQUEST: GET /api/v1/metrics/app 200"}
{"level":"info","ts":1643381514.9671488,"caller":"pktvisor/pktvisor.go:247","msg":"pktvisor stdout","log":"[2022-01-28 14:51:54.966] [visor] [info] REQUEST: GET /api/v1/taps 200"}
{"level":"info","ts":1643381514.967379,"caller":"agent/rpc_to.go:48","msg":"sending capabilities","value":"{\"schema_version\":\"1.0\",\"orb_agent\":{\"version\":\"0.12.0-develop\"},\"agent_tags\":null,\"backends\":{\"pktvisor\":{\"version\":\"3.4.0-develop-2da1810\",\"data\":{\"taps\":{\"default_pcap\":{\"config\":{\"iface\":\"wlo1\"},\"input_type\":\"pcap\",\"interface\":\"visor.module.input/1.0\"}}}}}}"}
{"level":"error","ts":1643381544.411644,"caller":"agent/logging.go:49","msg":"ERROR mqtt log","payload":["[client]  ","Connect comms goroutine - error triggered",{}],"stacktrace":"github.com/ns1labs/orb/agent.(*agentLoggerError).Println\n\t/go/src/github.com/ns1labs/orb/agent/logging.go:49\ngithub.com/eclipse/paho%2emqtt%2egolang.(*client).startCommsWorkers.func2\n\t/go/pkg/mod/github.com/eclipse/paho.mqtt.golang@v1.3.5/client.go:631"}
{"level":"error","ts":1643381549.2430634,"caller":"agent/comms.go:140","msg":"failed to subscribe to group channel/topic: time out","topic":"channels/d4b6667d-1ea5-4686-890c-bbee7d54297b/messages/fromcore","stacktrace":"github.com/ns1labs/orb/agent.(*orbAgent).subscribeGroupChannels\n\t/go/src/github.com/ns1labs/orb/agent/comms.go:140\ngithub.com/ns1labs/orb/agent.(*orbAgent).handleGroupMembership\n\t/go/src/github.com/ns1labs/orb/agent/rpc_from.go:20\ngithub.com/ns1labs/orb/agent.(*orbAgent).handleRPCFromCore\n\t/go/src/github.com/ns1labs/orb/agent/rpc_from.go:127\ngithub.com/eclipse/paho%2emqtt%2egolang.(*router).matchAndDispatch.func2\n\t/go/pkg/mod/github.com/eclipse/paho.mqtt.golang@v1.3.5/router.go:210"}
{"level":"error","ts":1643381549.2433925,"caller":"agent/logging.go:49","msg":"ERROR mqtt log","payload":["[net]     ","outgoing oboundp reporting error ",{}],"stacktrace":"github.com/ns1labs/orb/agent.(*agentLoggerError).Println\n\t/go/src/github.com/ns1labs/orb/agent/logging.go:49\ngithub.com/eclipse/paho%2emqtt%2egolang.startOutgoingComms.func1\n\t/go/pkg/mod/github.com/eclipse/paho.mqtt.golang@v1.3.5/net.go:317"}
{"level":"error","ts":1643381549.24355,"caller":"agent/logging.go:49","msg":"ERROR mqtt log","payload":["[client]  ","Connect comms goroutine - error triggered",{}],"stacktrace":"github.com/ns1labs/orb/agent.(*agentLoggerError).Println\n\t/go/src/github.com/ns1labs/orb/agent/logging.go:49\ngithub.com/eclipse/paho%2emqtt%2egolang.(*client).startCommsWorkers.func2\n\t/go/pkg/mod/github.com/eclipse/paho.mqtt.golang@v1.3.5/client.go:631"}
```

**Comment by manrodrigues** [comment]
@dscabral  I tested subscription and notice that:
1) the subscription to the group created is working, but
2) if the agent is already subscribed to a group and on agent editing I remove tag that matches with previous group and insert new tag matching with another group, although the subscription works, the agent is not unsubscribed from the other group

1 - create an agent with one tag matching an existing group
2 - create another group with different tag
3 - Edit the agent removing the first tag and inserting tag related to the second group

Expected result:
- agent must be subscribed to the second group
- agent must be unsubscribed from the first group

Current result:
- agent must be subscribed to the second group
**- agent is not unsubscribed from the first group**

**Comment by manrodrigues** [comment]
Ok

```
{"level":"info","ts":1644426012.5305712,"caller":"agent/comms.go:144","msg":"completed RPC subscription to group","group_id":"39b6675f-6094-424a-af28-80d758b790d1","group_name":"group","topic":"channels/d2984349-91bf-4ec3-bf76-46706dda85de/messages/fromcore"}
{"level":"info","ts":1644426072.4993114,"caller":"pktvisor/pktvisor.go:247","msg":"pktvisor stdout","log":"[2022-02-09 17:01:12.499] [visor] [info] REQUEST: GET /api/v1/policies/__all/metrics/bucket/1 200"}
{"level":"warn","ts":1644426072.4993598,"caller":"pktvisor/pktvisor.go:301","msg":"scrape: no policies found, skipping"}
{"level":"info","ts":1644426077.005432,"caller":"agent/comms.go:60","msg":"completed RPC unsubscription to group","group_id":"39b6675f-6094-424a-af28-80d758b790d1","group_name":"group","topic":"channels/d2984349-91bf-4ec3-bf76-46706dda85de/messages/fromcore"}
{"level":"info","ts":1644426077.0073206,"caller":"agent/comms.go:144","msg":"completed RPC subscription to group","group_id":"6a0c0f4e-8d7a-464c-8fc9-6301233bb168","group_name":"other_group","topic":"channels/3ca81aea-8c36-461d-947f-8b6587e2a3be/messages/fromcore"}
{"level":"warn","ts":1644426132.4993134,"caller":"pktvisor/pktvisor.go:301","msg":"scrape: no policies found, skipping"}
{"level":"info","ts":1644426132.4993083,"caller":"pktvisor/pktvisor.go:247","msg":"pktvisor stdout","log":"[2022-02-09 17:02:12.499] [visor] [info] REQUEST: GET /api/v1/policies/__all/metrics/bucket/1 200"}

```

![image](https://user-images.githubusercontent.com/78241475/153251647-5c1c89bf-e390-4917-9573-ec26d6e25c54.png)

**Comment by manrodrigues** [comment]
Steps to reproduce:
1 - provision an agent with tags
2 - create a group with the same tags
3 - Apply policies to this group
5 - Create another group with another tag
6 - Apply policies to this another group
7 - edit agent inserting same tags as both groups
8 - check container logs


Current result:

MQTT error and failed to subscribed

**Comment by manrodrigues** [comment]
Note that the policy is linked with agent, whilw the second group is not:

Policy-name is applied to group named "group": 

![image](https://user-images.githubusercontent.com/78241475/153627268-8f38ad02-9875-46fc-b99b-260dcf8a34c1.png)

Policy-compose is applied to group named "group2":

![image](https://user-images.githubusercontent.com/78241475/153627349-24ba731b-0731-4371-9025-08788155c265.png)


```
        },
        "group_state": {
            "54e9c416-e59c-4f8b-8813-e731f39b796e": {
                "channel": "9011c841-7269-4821-acfb-8c306c4bc75f",
                "name": "group"
            }
        },
        "policy_state": {
            "cf252daf-7e53-4dd1-853e-d8c18d866c79": {
                "datasets": [
                    "c166900c-56a1-44aa-87b9-8d586aa4cfad"
                ],
                "name": "policy-name",
                "state": "running"
            },
            "e5e6b5f7-3e0e-4534-8ed9-e2793a50f9a5": {
                "datasets": [
                    "4de76a7d-0416-4677-b05b-e7d293f83d28"
                ],
                "name": "policy-compose",
                "state": "running"
            }
        }
    }
```

**Comment by manrodrigues** [comment]
So, after restart the container errors still exist:

```
{"level":"info","ts":1644595264.2137778,"caller":"policyMgr/manager.go:175","msg":"policy applied successfully","policy_id":"cf252daf-7e53-4dd1-853e-d8c18d866c79","policy_name":"policy-name"}
{"level":"info","ts":1644595264.2138345,"caller":"policyMgr/manager.go:53","msg":"managing agent policy from core","action":"manage","name":"policy-compose","dataset":"4de76a7d-0416-4677-b05b-e7d293f83d28","backend":"pktvisor","id":"e5e6b5f7-3e0e-4534-8ed9-e2793a50f9a5","version":0}
{"level":"info","ts":1644595264.2153604,"caller":"pktvisor/pktvisor.go:247","msg":"pktvisor stdout","log":"[2022-02-11 16:01:04.215] [visor] [info] policy [policy-compose]: parsing"}
{"level":"info","ts":1644595264.2155857,"caller":"pktvisor/pktvisor.go:247","msg":"pktvisor stdout","log":"[2022-02-11 16:01:04.215] [visor] [info] policy [policy-compose]: instantiating Tap: default_pcap"}
{"level":"info","ts":1644595264.2159927,"caller":"pktvisor/pktvisor.go:247","msg":"pktvisor stdout","log":"[2022-02-11 16:01:04.215] [visor] [info] policy [policy-compose]: instantiating Handler handler_dns_1 of type dns"}
{"level":"info","ts":1644595264.2325046,"caller":"pktvisor/pktvisor.go:247","msg":"pktvisor stdout","log":"[2022-02-11 16:01:04.232] [visor] [info] policy [policy-compose]: starting"}
{"level":"info","ts":1644595264.2325609,"caller":"pktvisor/pktvisor.go:247","msg":"pktvisor stdout","log":"[2022-02-11 16:01:04.232] [visor] [info] policy [policy-compose]: starting handler instance: policy-compose-handler_dns_1"}
{"level":"info","ts":1644595264.232586,"caller":"pktvisor/pktvisor.go:247","msg":"pktvisor stdout","log":"[2022-02-11 16:01:04.232] [visor] [info] policy [policy-compose]: starting input instance: default_pcap-9ea395abea1cbc2d"}
{"level":"info","ts":1644595264.233098,"caller":"pktvisor/pktvisor.go:247","msg":"pktvisor stdout","log":"[2022-02-11 16:01:04.232] [visor] [info] REQUEST: POST /api/v1/policies 200"}
{"level":"info","ts":1644595264.233204,"caller":"policyMgr/manager.go:175","msg":"policy applied successfully","policy_id":"e5e6b5f7-3e0e-4534-8ed9-e2793a50f9a5","policy_name":"policy-compose"}
{"level":"error","ts":1644595294.2335806,"caller":"agent/heartbeats.go:71","msg":"error sending heartbeat","error":"publish was broken by timeout","stacktrace":"github.com/ns1labs/orb/agent.(*orbAgent).sendSingleHeartbeat\n\t/go/src/github.com/ns1labs/orb/agent/heartbeats.go:71\ngithub.com/ns1labs/orb/agent.(*orbAgent).handleAgentPolicies\n\t/go/src/github.com/ns1labs/orb/agent/rpc_from.go:34\ngithub.com/ns1labs/orb/agent.(*orbAgent).handleRPCFromCore\n\t/go/src/github.com/ns1labs/orb/agent/rpc_from.go:137\ngithub.com/eclipse/paho%2emqtt%2egolang.(*router).matchAndDispatch.func2\n\t/go/pkg/mod/github.com/eclipse/paho.mqtt.golang@v1.3.5/router.go:210"}
{"level":"error","ts":1644595294.2338252,"caller":"agent/logging.go:49","msg":"ERROR mqtt log","payload":["[net]     ","outgoing oboundp reporting error ",{}],"stacktrace":"github.com/ns1labs/orb/agent.(*agentLoggerError).Println\n\t/go/src/github.com/ns1labs/orb/agent/logging.go:49\ngithub.com/eclipse/paho%2emqtt%2egolang.startOutgoingComms.func1\n\t/go/pkg/mod/github.com/eclipse/paho.mqtt.golang@v1.3.5/net.go:317"}
{"level":"error","ts":1644595294.2339463,"caller":"agent/logging.go:49","msg":"ERROR mqtt log","payload":["[client]  ","Connect comms goroutine - error triggered",{}],"stacktrace":"github.com/ns1labs/orb/agent.(*agentLoggerError).Println\n\t/go/src/github.com/ns1labs/orb/agent/logging.go:49\ngithub.com/eclipse/paho%2emqtt%2egolang.(*client).startCommsWorkers.func2\n\t/go/pkg/mod/github.com/eclipse/paho.mqtt.golang@v1.3.5/client.go:631"}
{"level":"info","ts":1644595319.13533,"caller":"pktvisor/pktvisor.go:247","msg":"pktvisor stdout","log":"[2022-02-11 16:01:59.135] [visor] [warning] policy-compose-handler_dns_1 handler for policy policy-compose had a PeriodException, skipping: requested metrics period has not yet accumulated, current range is [0, 0]"}
{"level":"info","ts":1644595319.1353703,"caller":"pktvisor/pktvisor.go:247","msg":"pktvisor stdout","log":"[2022-02-11 16:01:59.135] [visor] [warning] policy-name-handler_dns_1 handler for policy policy-name had a PeriodException, skipping: requested metrics period has not yet accumulated, current range is [0, 0]"}
```

**Comment by manrodrigues** [comment]
If the tags are edited to unenroll the agent to some group, the group will be deleted from the agent but the policy will not... and if another dataset is created applying the policy that should have been removed to the remaining group, the log shows that the policy is already applied

**Comment by manrodrigues** [comment]
ok

```
{"level":"info","ts":1645766563.811989,"caller":"policyMgr/manager.go:183","msg":"policy applied successfully","policy_id":"8b3640b7-7d88-44c3-90bc-1842d50a45ef","policy_name":"policy3"}
{"level":"info","ts":1645766585.8116977,"caller":"pktvisor/pktvisor.go:247","msg":"pktvisor stdout","log":"[2022-02-25 05:23:05.811] [visor] [warning] policy3-handler_net_1 handler for policy policy3 had a PeriodException, skipping: requested metrics period has not yet accumulated, current range is [0, 0]"}
{"level":"info","ts":1645766585.812501,"caller":"pktvisor/pktvisor.go:247","msg":"pktvisor stdout","log":"[2022-02-25 05:23:05.812] [visor] [info] REQUEST: GET /api/v1/policies/__all/metrics/bucket/1 200"}
{"level":"info","ts":1645766585.8132386,"caller":"pktvisor/pktvisor.go:328","msg":"scraped metrics for policy","policy":"policy1","policy_id":"7653894b-fbfe-4ffb-9760-c17a7eb48931","payload_size_b":1637}
{"level":"info","ts":1645766585.8135018,"caller":"pktvisor/pktvisor.go:328","msg":"scraped metrics for policy","policy":"policy2","policy_id":"0ecfe150-634a-49b8-9a5f-233cc82ec0ff","payload_size_b":856}
{"level":"info","ts":1645766585.8161037,"caller":"pktvisor/pktvisor.go:346","msg":"scraped and published metrics","topic":"channels/4d809de5-5e3b-4092-b7eb-e8b5cf86d77e/messages/be/pktvisor/m/3","payload_size_b":2493,"batch_count":2}
{"level":"info","ts":1645766645.8111835,"caller":"pktvisor/pktvisor.go:247","msg":"pktvisor stdout","log":"[2022-02-25 05:24:05.811] [visor] [info] REQUEST: GET /api/v1/policies/__all/metrics/bucket/1 200"}
{"level":"info","ts":1645766645.8114703,"caller":"pktvisor/pktvisor.go:328","msg":"scraped metrics for policy","policy":"policy1","policy_id":"7653894b-fbfe-4ffb-9760-c17a7eb48931","payload_size_b":1692}
{"level":"info","ts":1645766645.811551,"caller":"pktvisor/pktvisor.go:328","msg":"scraped metrics for policy","policy":"policy2","policy_id":"0ecfe150-634a-49b8-9a5f-233cc82ec0ff","payload_size_b":901}
{"level":"info","ts":1645766645.8116221,"caller":"pktvisor/pktvisor.go:328","msg":"scraped metrics for policy","policy":"policy3","policy_id":"8b3640b7-7d88-44c3-90bc-1842d50a45ef","payload_size_b":899}
{"level":"info","ts":1645766645.8123462,"caller":"pktvisor/pktvisor.go:346","msg":"scraped and published metrics","topic":"channels/4d809de5-5e3b-4092-b7eb-e8b5cf86d77e/messages/be/pktvisor/m/3","payload_size_b":3492,"batch_count":3}

```
